skimage/filters/thresholding.py

```diff
@@ -578,8 +578,7 @@ def threshold_li(image, *, tolerance=None):
     # Li's algorithm requires positive image (because of log(mean))
     image_min = np.min(image)
     image -= image_min
-    image_range = np.max(image)
-    tolerance = tolerance or 0.5 * image_range / 256
+    tolerance = tolerance or np.min(np.diff(np.unique(image))) / 2
```

Comment:
From above:
>By default, this is the smallest difference between intensity values in ``image``.

Here, it is a half of that difference.

Reply:
@soupault thanks, fixed!